### DISCUSSION TITLE
Carbon Aware Azure Functions

### DISCUSSION THREAD

**Discussion by georgekosmidis** [Green-Software-Foundation/carbon-aware-sdk#304] [General]
Hello all,

I ve been working on a small piece of code that brings the Carbon Aware SDK to Azure Functions (you can check it here: https://github.com/georgekosmidis/Greenhopper).

I ve been pulling Carbon Aware SDK repo as a submodule and using parts of it in my solution as a code reference. This works well for developing but when I am trying to pack it as nuget with `-IncludeReferencedProjects`  i get the following error (upon referencing the nuget package):

`Assembly 'Greenhopper' with identity 'Greenhopper, Version=0.3.4.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.Extensions.Hosting.Abstractions, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.Extensions.Hosting.Abstractions' with identity 'Microsoft.Extensions.Hosting.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'`

Is there a chance we could somehow align on the versions, at least up until there is nuget package?

Thanks!

**Comment by danuw** [comment]
Interesting. Thank you for pointing this out. Would you be able to create an issue about this and tag me please? Will try and bring up at next catch up

**Comment by danuw** [comment]
making sure this is linked https://github.com/Green-Software-Foundation/carbon-aware-sdk/issues/306
